R/make_map.r

```diff
@@ -8,25 +8,44 @@
 #' @export
 #' @importFrom magrittr "%>%"
 #'
-make_map <- function(data = NULL, map = NULL, utm33 = FALSE) {
+make_map <- function(data = NULL, map = NULL, utm33 = TRUE) {
 
   output <- NULL
-
-  simple_data <- data[c("area", "value")]
-  map_data <- merge(x = map, y = simple_data, by.x = "area_num", by.y = "area")
-  map_data$area_num <- NULL
-  map_data$area_name <- NULL
-
   if (utm33) {
     # convert from utm33 to leaflet
-    map_data <- kart::utm33_to_leaflet(map = map_data, sf = TRUE)
+    map_data <- kart::utm33_to_leaflet(map = map, sf = TRUE)
   }
-
-  output <- graphics::plot(map_data,
-                           main = "",
-                           breaks = "quantile", nbreaks = 4,
-                           pal = SKDEr::skde_colors(num = 4)
-                 )
-
+  
+  map_data <- methods::as(map_data, Class = "Spatial")
+  simple_data <- data[c("area", "value")]
+  map_data@data <- map_data@data %>% dplyr::left_join(simple_data, by = c("area_num" = "area"))
+  pal <- leaflet::colorBin(palette = SKDEr::skde_colors(num=5), 
+                           domain = map_data@data$value, 
+                           bins = 5, 
+                           pretty = FALSE)
+  
+  output <- 
+    map_data %>% 
+    leaflet::leaflet(options = leaflet::leafletOptions(minZoom = 5, maxZoom = 5)) %>% 
+    leaflet::addPolygons(weight = 1, 
+                color  = "black",
+                fillColor = ~pal(value),
+                fillOpacity = 0.8,
+                label = base::sprintf("<strong>%s</strong></br> %g", 
+                                      map_data@data$area_name, 
+                                      round(map_data@data$value, 0)) %>% lapply(htmltools::HTML),  
```

Comment:
This is where it fails. I will try to rewrite. There is no `area_name` in `map@data` as far as I can see.